tests/unittests/segmentation/test_mean_iou.py

```diff
@@ -99,3 +99,66 @@ def test_mean_iou_functional(self, preds, target, input_format, include_backgrou
                 "input_format": input_format,
             },
         )
+
+
+class TestMeanIoUWithoutNumClasses(MetricTester):
```

Comment:
Instead of creating a completely new test class, lets just extend the current one. In the parametrize
https://github.com/Lightning-AI/torchmetrics/blob/e1955a42148c844e8efcc9a824952d66be97a67c/tests/unittests/segmentation/test_mean_iou.py#L48-L55
add an additional argument `num_classes` and then a new testcase with `one-hot` and `num_classes=None`